install.sh

```diff
@@ -144,6 +144,16 @@ Install()
     # If update, start Wazuh
     if [ "X${update_only}" = "Xyes" ]; then
         WazuhUpgrade $INSTYPE
+
+        # Update versions previous to Wazuh 4.8
+        local OSSEC_CONFIGURATION_FILE="$USER_DIR/etc/ossec.conf"
+        local VULN_TEMPLATE="./etc/templates/config/generic/wodle-vulnerability-detection.manager.template"
+        local INDEXER_TEMPLATE="./etc/templates/config/generic/wodle-indexer.manager.template"
+
+        # Load vulnerability detector replacer commands
+        . ./src/init/replace_vulnerability_detector.sh
+        updateVulnerabilityDetector $OSSEC_CONFIGURATION_FILE $VULN_TEMPLATE $INDEXER_TEMPLATE
```

Comment:
I think the installation type check should be done here or, alternatively, the installation type should be also passed to the called function.

If we use that function outside `install.sh`, we won't be sure that `INSTYPE` type will be accessible.

```suggestion
        if [ "$INSTYPE" = "server" ]; then
                # Load vulnerability detector replacer commands
                . ./src/init/replace_vulnerability_detector.sh
                updateVulnerabilityDetector $OSSEC_CONFIGURATION_FILE $VULN_TEMPLATE $INDEXER_TEMPLATE
        fi
```

Reply:
I agree with this. On top of that, when using sh, the string comparisons are made by prepending and `X` at the beginning (or any character)

https://github.com/wazuh/wazuh/blob/bcce944193f5767f051119cd02e5e24af65e6a17/install.sh#L136

This is due to [an error](https://www.unix.com/shell-programming-and-scripting/141856-how-avoid-unexpected-operator-error-when-comparing-2-strings.html) in the shell

Reply:
Could be tested with this script
```bash
#!/bin/sh
a=''
c='testing'

if [ "$a" == "$c" ]; then
  echo "$a = $c"
else
  echo "They are not equal"
fi
```

```
./a.sh 
./a.sh: 5: [: unexpected operator
They are not equal
```

Reply:
@tsarquis88 I moved the check to the `install.sh` script

@sebasfalcone I added the `X` in the comparison